felix/bpf/failsafes/failsafes.go

```diff
@@ -110,25 +117,46 @@ func (m *Manager) ResyncFailsafes() error {
 			return
 		}
 
-		ipv4 := ip.To4()
-		if ipv4 == nil || len(ipv4) != 4 {
-			// If ipv4 is nil, then the IP is not an IPv4 address. Only IPv4 addresses are supported in failsafes.
-			log.Errorf("Invalid IPv4 address configured in the failsafe ports: %s", cidr)
-			syncFailed = true
+		if ipnet.Version() != int(m.ipFamily) {
 			return
 		}
 
 		mask, bits := ipnet.Mask.Size()
-		if bits != 32 {
-			log.Errorf("CIDR mask size not valid for IPv4 addresses: %d", bits)
-			syncFailed = true
-			return
+		maskedIPStr := ""
+		if m.ipFamily == proto.IPVersion_IPV4 {
+			ipv4 := ip.To4()
+			if ipv4 == nil || len(ipv4) != 4 {
+				// If ipv4 is nil, then the IP is not an IPv4 address. Only IPv4 addresses are supported in failsafes.
+				log.Errorf("Invalid IPv4 address configured in the failsafe ports: %s", cidr)
+				syncFailed = true
+				return
+			}
+
+			if bits != 32 {
+				log.Errorf("CIDR mask size not valid for IPv4 addresses: %d", bits)
+				syncFailed = true
+				return
+			}
```

Comment:
Isnt that already handled by the fact that `if ipnet.Version() == int(m.ipFamily)` ? We did not have that check before.